tdp/core/variables/service_variables.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2022 TOSIT.IO
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from collections import OrderedDict
+from contextlib import ExitStack, contextmanager
+from pathlib import Path
+
+from tdp.core.collection import YML_EXTENSION
+from tdp.core.operation import Operation
+from tdp.core.variables.variables import Variables, merge_hash
+
+logger = logging.getLogger("tdp").getChild("service_variables")
+
+SERVICE_NAME_MAX_LENGTH = 15
+
+
+class ServiceVariables:
+    def __init__(self, service_name, repository):
+        if len(service_name) > SERVICE_NAME_MAX_LENGTH:
+            raise ValueError(f"{service_name} is longer than {SERVICE_NAME_MAX_LENGTH}")
+        self._name = service_name
+        self._repo = repository
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def repository(self):
+        return self._repo
+
+    @property
+    def version(self):
+        return self.repository.current_version()
+
+    @property
+    def clean(self):
+        return self.repository.is_clean()
+
+    @property
+    def path(self):
+        return self.repository.path
+
+    # TODO: move this function outside of this class, or move the dag part
+    def get_component_name(self, dag, component):
+        operations_filtered = list(
+            filter(
+                lambda operation: operation.component == component,
+                dag.services_operations[self.name],
+            )
+        )
+        if operations_filtered:
+            operation = operations_filtered[0]
+            return self.name + "_" + operation.component
+        raise ValueError(f"Service {self.name} does not have a component {component}")
+
+    def update_from_variables_folder(self, message, tdp_vars_overrides):
+        override_files = list(tdp_vars_overrides.glob("*" + YML_EXTENSION))
+        service_files_to_open = [override_file.name for override_file in override_files]
+        with self.open_var_files(
+            f"{self.name}: {message}", service_files_to_open
+        ) as configurations:
+            for file in override_files:
+                logger.info(f"Initializing {self.name} with variables from {file}")
```

Comment:
It is always an "init" ? When multiple `update_from_variables_folder` are called to add multiple collection and overrides, it will merge with existing and it is no longer an "init" no ?

Reply:
You're right, done.